src/entities/User.ts

```diff
@@ -0,0 +1,30 @@
+import { Field, ID, ObjectType } from 'type-graphql';
+import { prop as Property, DocumentType, getModelForClass } from '@typegoose/typegoose';
+import mongoose from 'mongoose';
+import { PublicationID } from '../common/types';
+
+@ObjectType({ description: 'The User Model' })
+export class User {
+  @Field(() => ID)
+  id: string;
+
+  @Field()
+  @Property()
+  uuid: string; // id of user stored on frontend
+
+  // https://github.com/typegoose/typegoose/issues/522
+  // https://github.com/typegoose/typegoose/discussions/380
+  @Field((type) => [PublicationID])
+  @Property({ required: true, type: () => PublicationID, default: [] })
+  followedPublications: mongoose.Types.DocumentArray<DocumentType<PublicationID>>;
+
+  @Field()
+  @Property()
+  deviceToken: string;
+
+  @Field()
+  @Property()
+  deviceType: string; // ANDROID, IOS
```

Comment:
I think the extra storage space is negligible. I also think its more intuitive to distinguish using some keyword. If we decided to add logic to support desktop notifications down the line, we wouldn't be able to use a boolean. An "deviceType" enum type is probably the best structure to use but I think string is fine for now to avoid issues with GraphQL types.